### ISSUE TITLE
[core] Implement runtime plugins for additional package managers (mamba, micromamba, pixi, etc.)

### ISSUE THREAD

**Issue by cynddl** [ray-project/ray#45572]
### Description

Opening this issue to see if there's interest to add other package managers for runtime environments? The conda plugin is fairly big since it needs to create new environments—but I'd love to have a very simple plugin to activate a virtual environment from e.g. mamba, micromamba, or pixi.

Despite conda now using libmamba (as mentioned in #33596), many people use other package managers. Not having to play with fire (I had to set ‘alias conda=mamba’) to use ray would be much appreciated. This seems like a shame since using any environment, from any package manager, is a fairly simple task with a small runtime plugin with _modify_context_.

I'd be happy to contribute a small plugin for pixi (https://github.com/prefix-dev/pixi).

### Use case

There has been much progress in the development of package managers since (ana/mini)conda. Some of them are much smaller and easier to install on HPC (micromamba), others allow the centralised development of multiple conda+pip environments (pixi). Having the option to support running workers in these environments would maximise usability of ray outside conda.

**Comment by cynddl** [comment]
Just wanted to kindly bump this issue. @anyscalesam @jjyao, is there anything I can do to help here? Would this contribution be of interest?

**Comment by cynddl** [comment]
Another small bump as I see that uv support has been added (#47819). Any interest in supporting pixi too? I have a patch that works well locally, happy to share it.

**Comment by ljstrnadiii** [comment]
@cynddl would this enable support for something like `ray job submit --working-dir . -- pixi run test.py` as explained [here](https://docs.ray.io/en/latest/ray-core/handling-dependencies.html#using-uv-for-package-management)? If so, I think it is only a matter of time until this would be in high demand!

**Comment by ljstrnadiii** [comment]
I've decided to use pixi and have been struggling to get the full env activated on workers. @cynddl do you have a patch to share? I'm curious what this would look like.

**Comment by cynddl** [comment]
@ljstrnadiii Of course! This is what I have, which works well for my usage (but might be different from yours with ray job): https://gist.github.com/cynddl/13797be88da5e5b882f37c8876f8cf25

**Comment by cynddl** [comment]
I typically use it with:

```python
os.environ["RAY_RUNTIME_ENV_PLUGINS"] = '[{"class": "PACKAGENAME.ray_plugins.pixi:PixiPlugin"}]'
```

**Comment by geoHeil** [comment]
For me the situation is different:

I am on a local ray cluster.

- Importing the pixi editable pip package works fine (from both ray + bash)
- Importing fails when I try to orchestrate my ray script from dagster https://github.com/danielgafni/dagster-ray - then the import is no longer found.

I could not fix the import with this pixi plugin though.
